CHANGELOG.md

```diff
@@ -1,3 +1,11 @@
+## 3.1.1
```

Comment:
are all these changes worthy of a minor bump maybe? any potential BWC issues?

Reply:
Actually I think you are correct. There are no known BWC issues but there is a new feature.